syntax = "proto3";

service Booking {
    rpc GetBookingByUserID(UserID) returns (BookingData) {}
    rpc AddBooking(BookingRequest) returns (BookingResponse) {}
    rpc GetAllBookings(BookingEmpty) returns (stream BookingData) {}
}

message UserID {
    string userid = 1;
}

message BookingEmpty {}

message BookingRequest {
    string userid = 1;
    string date = 2;
    string movieid = 3;
}

message BookingData {
  string userid = 1;
  repeated DateEntry dates = 2;
}

message DateEntry {
    string date = 1;
    repeated string movies = 2;
}

message BookingResponse {
    string message = 1;
    string error = 2;
}
